src/providers/list/hooks/useProvidersInventoryList.ts

```diff
@@ -0,0 +1,125 @@
+import { useEffect, useRef, useState } from 'react';
+
+import type { ProvidersInventoryList, V1beta1Provider } from '@kubev2v/types';
+import { consoleFetchJSON, useFlag } from '@openshift-console/dynamic-plugin-sdk';
+
+import { getInventoryApiUrl } from '../../../modules/Providers/utils/helpers/getApiUrl';
+import { DEFAULT_FIELDS_TO_AVOID_COMPARING } from '../utils/constants';
+import { k8sGetProvidersByNamespace } from '../utils/k8sGetProvidersByNamespace';
+import { updateInventoryIfChanged } from '../utils/updateInventoryIfChanged';
+
+type UseInventoryParams = {
+  namespace?: string;
+  interval?: number; // Polling interval in milliseconds
+};
+
+type UseInventoryResult = {
+  inventory: ProvidersInventoryList | null;
+  loading: boolean;
+  error: Error | null;
+};
+
+const useProvidersInventoryList = ({
+  interval = 20000,
+  namespace = undefined,
+}: UseInventoryParams): UseInventoryResult => {
+  const [inventory, setInventory] = useState<ProvidersInventoryList | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [error, setError] = useState<Error | null>(null);
+  const oldDataRef = useRef<ProvidersInventoryList | null>(null);
+  const oldErrorRef = useRef<string | null>(null);
+  const canList: boolean = useFlag('CAN_LIST_NS');
+
+  const getInventoryByNamespace = async (
+    currNamespace: string | undefined,
+  ): Promise<ProvidersInventoryList | null> => {
+    const newInventory: ProvidersInventoryList = {
+      openshift: [],
+      openstack: [],
+      ova: [],
+      ovirt: [],
+      vsphere: [],
+    };
+
+    const providers = await k8sGetProvidersByNamespace(currNamespace);
+
+    const readyProviders = providers?.filter(
+      (provider: V1beta1Provider) => provider?.status?.phase === 'Ready',
+    );
+
+    const inventoryProviderURL = (provider: V1beta1Provider) =>
+      `providers/${provider?.spec?.type}/${provider?.metadata?.uid}`;
+
+    const allPromises = Promise.all(
+      readyProviders.map(async (provider) => {
```

Comment:
u are mixing async and native promise with then

Reply:
I declared an async function that returns `Promise.all` for fetching all inventory providers together. Why it's not recommended?